dotnet-test-samples/async-architectures/async-lambda-dynamodb/src/AsyncTesting.S3EventHandler/Function.cs

```diff
@@ -0,0 +1,111 @@
+using Amazon.Lambda.Core;
+using Amazon.Lambda.S3Events;
+using Amazon.S3;
+using Amazon.S3.Util;
+
+// Assembly attribute to enable the Lambda function's JSON input to be converted into a .NET class.
+[assembly: LambdaSerializer(typeof(Amazon.Lambda.Serialization.SystemTextJson.DefaultLambdaJsonSerializer))]
+
+namespace AsyncTesting.S3EventHandler;
+
+using System.Text;
+
+using Amazon.S3.Model;
+using Amazon.XRay.Recorder.Handlers.AwsSdk;
+
+using AWS.Lambda.Powertools.Logging;
+using AWS.Lambda.Powertools.Tracing;
+
+public class Function
+{
+    private IAmazonS3 _s3Client;
+
+    /// <summary>
+    /// Default constructor. This constructor is used by Lambda to construct the instance. When invoked in a Lambda environment
+    /// the AWS credentials will come from the IAM role associated with the function and the AWS region will be set to the
+    /// region the Lambda function is executed in.
+    /// </summary>
+    public Function() : this(null)
+    {
+        AWSSDKHandler.RegisterXRayForAllServices();
+    }
+
+    /// <summary>
+    /// Constructs an instance with a preconfigured S3 client. This can be used for testing outside of the Lambda environment.
+    /// </summary>
+    /// <param name="s3Client"></param>
+    public Function(IAmazonS3? s3Client)
```

Comment:
Why use nullable? this c'tor would only be called in tests

Reply:
The primary constructor also calls it, using a null value. Line 28 in this same file